src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -261,280 +212,337 @@ class BasePaymentsPage extends React.Component {
+                                popoverPlacement="bottom"
+                            >
+                                {(triggerKYCFlow) => (
+                                    <MenuItem
+                                        title={translate('common.transferBalance')}
+                                        icon={Expensicons.Transfer}
+                                        onPress={triggerKYCFlow}
+                                        shouldShowRightIcon
+                                        disabled={props.network.isOffline}
+                                    />
+                                )}
+                            </KYCWall>
+                        </View>
+                    )}
+                </>
+            )}
+            <Text style={[styles.ph5, styles.textLabelSupporting, styles.mb1]}>{translate('paymentsPage.paymentMethodsTitle')}</Text>
+        </>
+    );
+
+    useEffect(() => {
+        PaymentMethods.openPaymentsPage();
+    }, []);
+
+    useEffect(() => {
+        // If the user was previously offline, skip debouncing showing the loader
+        if (!props.network.isOffline) {
+            updateShouldShowLoadingSpinner();
+        } else {
+            debounceSetShouldShowLoadingSpinner();
+        }
+    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
+
+    useEffect(() => {
+        if (props.shouldListenForResize) {
+            setMenuPosition();
+        }
+
+        if (shouldShowDefaultDeleteMenu || showPassword.shouldShowPasswordPrompt) {
+            // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
+            let shouldResetPaymentMethodData = false;
+
+            if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(props.bankAccountList[paymentMethod.methodID])) {
+                shouldResetPaymentMethodData = true;
+            } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(props.cardList[paymentMethod.methodID])) {
+                shouldResetPaymentMethodData = true;
+            } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && _.isEmpty(props.payPalMeData)) {
+                shouldResetPaymentMethodData = true;
+            }
+            if (shouldResetPaymentMethodData) {
+                // Close corresponding selected payment method modals which are open
+                if (shouldShowDefaultDeleteMenu) {
+                    hideDefaultDeleteMenu();
+                } else if (showPassword.shouldShowPasswordPrompt) {
+                    hidePasswordPrompt();
+                }
+            }
+        }
+
+        // previously online OR currently offline, skip fetch
+        if (props.network.isOffline) {
+            return;
+        }
+        PaymentMethods.openPaymentsPage();
```

Comment:
Not sure why this should be clubbed here along with other dependencies. If other Items in this effect changes and  PaymentMethods.openPaymentsPage() will be executed it's unnecessary. This should specifically happen when previous offline now become online, please the old was a reference. I think the `useEffect`  can be simplified into multiple separate using only required dependencies.

Reply:
Yeah, you're right.
Updated following your instruction.